test/gzip.js

```diff
@@ -56,6 +59,11 @@ test('handles gzip error', async t => {
 	t.is(err.name, 'ReadError');
 });
 
+test('decompress option opts out of decompressing', async t => {
+	const response = await got(s.url, {decompress: false, encoding: null});
```

Comment:
Woops, sorry, completely missed that.

Encoding should be set to null internally if `decompress` is `false`. I'll update.